neo/app/actor/NeoActor.scala

```diff
@@ -2,74 +2,95 @@ package actor
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import akka.stream.Materializer
-import akka.util.{ByteString, _}
 import akka.stream.scaladsl._
+import akka.util.{ByteString, Timeout}
 import edu.uci.ics.cloudberry.zion.common.Config
 import edu.uci.ics.cloudberry.zion.model.schema.TimeField
 import models.{GeoLevel, UserRequest}
 import org.joda.time.DateTime
-import play.api.Logger
+import play.Logger
 import play.api.libs.json._
 import play.api.libs.ws._
 
-import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
 
 class NeoActor(out: ActorRef, ws: WSClient, host: String, config: Config)
               (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
 
   import NeoActor._
   import RequestType._
+
   implicit val timeout: Timeout = Timeout(20.minutes)
+  val url : String = "http://" + host + "/berry"
 
   override def receive: Receive = {
     case json: JsValue =>
-      json.validate[UserRequest].map { userRequest =>
-        val berryRequest = generateCBerryRequest(userRequest)
-        val url = "http://" + host + "/berry"
-        handleSamplingResponse(url, berryRequest)
-        handleSlicingResponse(url, berryRequest)
-      }.recoverTotal {
-        e => out ! JsError.toJson(e)
+      if ((json \ "cmd").asOpt[String].contains(RequestType.TotalCount.toString)) {
+        val dataset = (json \ "dataset").asOpt[String].getOrElse("twitter.ds_tweet")
+        val berryJson = generateCountRequest(dataset)
+        handleCountResponse(berryJson)
+        context.system.scheduler.schedule(1 seconds, 1 seconds, self, TotalCountRequest(berryJson))
+      } else {
+        json.validate[UserRequest].map { userRequest =>
+          val berryRequest = generateCBerryRequest(userRequest)
+          handleSamplingResponse(berryRequest)
+          handleSlicingResponse(berryRequest)
+        }.recoverTotal {
+          e =>
+            out ! JsError.toJson(e)
+        }
       }
+    case r: TotalCountRequest =>
+      handleCountResponse(r.berryJson)
+    case x => log.error("unknown:" + x)
```

Comment:
Changed to `Logger.error`. 
I found previously we are using `play.api.Logger` which won't work. We actually should use `play.Logger` which showed the expected log. Don't know why they have two identical things.